pennylane/ops/op_math/controlled.py

```diff
@@ -127,19 +127,29 @@ def cond_fn():
         ops_loader = available_eps[active_jit]["ops"].load()
         return ops_loader.ctrl(op, control, control_values=control_values, work_wires=work_wires)
 
-    custom_controlled_ops = {
+    non_parametric_custom_ops = {
         (qml.PauliZ, 1): qml.CZ,
         (qml.PauliY, 1): qml.CY,
         (qml.PauliX, 1): qml.CNOT,
         (qml.PauliX, 2): qml.Toffoli,
     }
+    parametric_custom_ops = {
+        (qml.RX, 1): qml.CRX,
+        (qml.RY, 1): qml.CRY,
+        (qml.RZ, 1): qml.CRZ,
+        (qml.Rot, 1): qml.CRot,
+        (qml.PhaseShift, 1): qml.ControlledPhaseShift,
+    }
     control_values = [control_values] if isinstance(control_values, (int, bool)) else control_values
     control = qml.wires.Wires(control)
     custom_key = (type(op), len(control))
 
-    if custom_key in custom_controlled_ops and (control_values is None or all(control_values)):
+    if custom_key in non_parametric_custom_ops and (control_values is None or all(control_values)):
+        qml.QueuingManager.remove(op)
+        return non_parametric_custom_ops[custom_key](control + op.wires)
+    if custom_key in parametric_custom_ops and (control_values is None or all(control_values)):
         qml.QueuingManager.remove(op)
-        return custom_controlled_ops[custom_key](control + op.wires)
+        return parametric_custom_ops[custom_key](*op.data, control + op.wires)
```

Comment:
We might want to extract this into a helper function at some point in the future, but that would be out of scope here.

Reply:
I wonder if we can simply reuse the `_controlled` method here. Something like:
```
if hasattr(op, "_controlled") and (control_values is None or all(control_values)):
    qml.QueuingManager.remove(op)
    return op._controlled(control_wires)
```
What was the `_controlled` method originally created for? I'm confused why it isn't used here.

Reply:
do you need to make separate dicts and handlers for parametric vs. non-parametric ops? in the non-parametric case, `*op.data` will just be nothing, so you can pass it to the constructor without any unintended consequence

Reply:
> do you need to make separate dicts and handlers for parametric vs. non-parametric ops? in the non-parametric case, `*op.data` will just be nothing, so you can pass it to the constructor without any unintended consequence

Good point. I'll make that change. I don't know why I did that.